src/llama_stack/providers/inline/agents/meta_reference/responses/streaming.py

```diff
@@ -1159,6 +1186,149 @@ async def _process_tools(
                 ):
                     yield stream_event
 
+    async def _process_tool_choice(self) -> str | dict[str, Any] | None:
```

Comment:
this should be a "free" function -- should not need any "self" state semantically